src/goto-instrument/contracts/contracts.h

```diff
@@ -33,6 +33,10 @@ Date: February 2016
 #define HELP_LOOP_CONTRACTS                                                    \
   " {y--apply-loop-contracts} \t check and use loop contracts when provided\n"
 
+#define FLAG_DISABLE_SIDE_EFFECT_CHECK "disable-side-effect-check"
+#define HELP_DISABLE_SIDE_EFFECT_CHECK                                         \
+  " {y--disable-side-effect-check} \t UNSOUND OPTION. do not check \t "        \
```

Comment:
We should be able to instrument the GOTO program snippet that encodes the invariant evaluation using an empty write set and still check for side-effect freeness, as is done for contracts pre and post conditions.

Reply:
As the comment above, this flag allow us to disable side-effect checks while we develop the new checks.